src/main/java/appeng/util/ConfigInventory.java

```diff
@@ -207,8 +207,10 @@ public void setStack(int slot, @Nullable GenericStack stack) {
                 // force amount to 0 in types-only mode
                 stack = new GenericStack(stack.what(), 0);
             } else if (!typesOnly && stack.amount() <= 0) {
-                // in stack mode, amounts of 0 or less clear the slot
-                stack = null;
+                // in stack mode, amounts of 0 or less will:
+                // - filter the slot to a minimum of 1 of that stack, if empty
+                // - clear the slot, if not empty
+                stack = getStack(slot) == null && mode == Mode.CONFIG_STACKS ? new GenericStack(stack.what(), 1) : null;
```

Comment:
A bit cleaner I think, and with a comment giving an example:
```suggestion
                if (mode == Mode.CONFIG_STACKS && getStack(slot) == null) {
                    // filter the slot to a minimum of 1 of that stack, if currently empty
                    // This allows setting a filter using empty containers that are limited to one resource, e.g. empty mana tablet.
                    stack = new GenericStack(stack.what(), 1);
                } else {
                    // in stack mode, amounts of 0 or less clear the slot
                    stack = null;
                }
```